src/main/java/io/reactivex/Flowable.java

```diff
@@ -7669,9 +7669,24 @@ public final void blockingSubscribe(Subscriber<? super T> subscriber) {
     }
 
     /**
-     * Returns a Flowable that emits all items emitted by the source Publisher that are distinct.
+     * Returns a Flowable that emits all items emitted by the source Publisher that are distinct
+     * based on {@link Object#equals(Object)} comparison.
      * <p>
      * <img width="640" height="310" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/distinct.png" alt="">
+     * <p>
+     * It is recommended the elements' class {@code T} in the flow overrides the default {@code Object.equals()} to provide
+     * meaningful comparison between items as the default Java implementation only considers reference equivalence.
+     * <p>
+     * By default, {@code distinct()} uses an internal, per Subscriber {@link java.util.HashSet} to remember
```

Comment:
supernit: could be phrased like that `By default, {@code distinct()} uses an internal {@link java.util.HashSet} per Subscriber to remember`

Reply:
Same below